compiler/rustc_mir/src/transform/check_consts/validation.rs

```diff
@@ -556,7 +582,13 @@ impl Visitor<'tcx> for Validator<'mir, 'tcx> {
                 );
 
                 if borrowed_place_has_mut_interior {
-                    self.check_op(ops::CellBorrow);
+                    // Locals without StorageDead follow the "enclosing scope" rule, meaning
+                    // they are essentially anonymous static items themselves.
+                    if self.local_has_storage_dead(place.local) {
+                        self.check_op(ops::CellBorrowBehindRef);
```

Comment:
Why "BorrowBehindRef"? I cannot make sense of this name.

Reply:
Uh... that's a naming scheme I never updated. I only fixed it in the `&mut` PR, not here, will do